plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -40,169 +42,202 @@ export default function Content(): JSX.Element {
-	}, [] );
-
-	// Get the content for this screen
-	useEffect( () => {
-		const abortController = new AbortController();
-
-		if (
-			query.tab === undefined ||
-			( query.tab && [ '', 'discover' ].includes( query.tab ) )
-		) {
-			return;
-		}
-
 		setIsLoading( true );
-		setProducts( [] );
-
-		const params = new URLSearchParams();
-
-		if ( query.term ) {
-			params.append( 'term', query.term );
-		}
+		setAllProducts( [] );
 
-		if ( query.category ) {
-			params.append(
-				'category',
-				query.category === '_all' ? '' : query.category
-			);
-		} else if ( query?.tab === 'themes' ) {
-			params.append( 'category', 'themes' );
-		} else if ( query?.tab === 'business-services' ) {
-			params.append( 'category', 'business-services' );
-		} else if ( query?.tab === 'search' ) {
-			params.append( 'category', 'extensions-themes-business-services' );
-		}
+		// If query.category is present and not '_all', only fetch that category
+		if ( query.category && query.category !== '_all' ) {
+			const params = new URLSearchParams();
+			params.append( 'category', query.category );
 
-		const wccomSettings = getAdminSetting( 'wccomHelper', false );
-		if ( wccomSettings.storeCountry ) {
-			params.append( 'country', wccomSettings.storeCountry );
-		}
+			if ( query.term ) {
+				params.append( 'term', query.term );
+			}
 
-		fetchSearchResults( params, abortController.signal )
-			.then( ( productList ) => {
-				setProducts( productList );
+			const wccomSettings = getAdminSetting( 'wccomHelper', false );
+			if ( wccomSettings.storeCountry ) {
+				params.append( 'country', wccomSettings.storeCountry );
+			}
 
-				if ( query.term ) {
+			fetchSearchResults( params, abortControllers[ 0 ].signal )
+				.then( ( productList ) => {
+					setAllProducts( productList );
 					setSearchResultsCount( {
 						extensions: productList.filter(
-							( p ) => p.type === 'extension'
+							( p ) => p.type === ProductType.extension
 						).length,
 						themes: productList.filter(
-							( p ) => p.type === 'theme'
+							( p ) => p.type === ProductType.theme
 						).length,
 						'business-services': productList.filter(
-							( p ) => p.type === 'business-service'
+							( p ) => p.type === ProductType.businessService
 						).length,
 					} );
-				}
-			} )
-			.catch( () => {
-				setProducts( [] );
-			} )
-			.finally( () => {
-				// we are recording both the new and legacy events here for now
-				// they're separate methods to make it easier to remove the legacy one later
-				const marketplaceViewProps = {
-					view: query?.tab,
-					search_term: query?.term,
-					product_type: query?.section,
-					category: query?.category,
-				};
-
-				recordMarketplaceView( marketplaceViewProps );
-				recordLegacyTabView( marketplaceViewProps );
-				setIsLoading( false );
-			} );
+				} )
+				.catch( () => {
+					setAllProducts( [] );
+				} )
+				.finally( () => {
+					setIsLoading( false );
+				} );
+		} else {
```

Comment:
I'm not a massive fan of big `else` statements. They might be difficult to follow and this else block has a lot going on. 

As a first step, would it make sense to create a new `fetchSearchResultsForAllCategories` helper function that will include the `Promise.all` code to reduce the code? Just as a brainstorming execise

Reply:
This is a really good catch! I also didn't like the blocks within this conditional. As testing steps are complex and since this PR is blocking several other PRs, I will merge as is. Created https://github.com/Automattic/woocommerce.com/issues/21700 to address this and other refactor suggestions.